src/rules/maxLineLengthRule.ts

```diff
@@ -45,19 +73,31 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public isEnabled(): boolean {
-        return super.isEnabled() && this.ruleArguments[0] as number > 0;
+        const argument = this.ruleArguments[0];
+        const numberGreaterThan0 = (Number(argument) > 0);
+        const limitGreaterTHan0 = (argument instanceof Object && argument.limit > 0);
+        return super.isEnabled() && (numberGreaterThan0 || limitGreaterTHan0);
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        const argument = this.ruleArguments[0];
+        const options = {
+            ignorePattern: (typeof argument === "object") ? new RegExp(argument["ignore-pattern"]) : undefined,
```

Comment:
you also need to check if `argument["ignore-pattern"]` is present because `new RegExp(undefined)` matches everything